templates/static/js/example.js

```diff
@@ -38,32 +80,72 @@
     [].slice.call(examples).forEach(fetchExample);
   });
 
-  function fetchExample(exampleElement) {
-    var link = exampleElement.href;
+  /**
+   * Sends a `GET` request to `url` to request an example's contents.
+   * @param {String} url Address of the example
+   * @returns {Promise<String>} Response text
+   */
+  async function fetchExampleResponseText(url) {
```

Comment:
This PR involved adding an additional network request for Jinja examples, so I extracted our network fetching code to its own async function to make it easier to await multiple network calls.

What you're seeing here is my attempt to "promisify" [XMLHttpRequest](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest), rather than change to [Fetch](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch).  The returned promises are `try`/`catch` handled in `fetchExample`. 

That being said, this would be a lot simpler if we used `fetch()`. I wonder if there's some reason we used `XMLHttpRequest` rather than Fetch (perhaps it's the lack of IE support, or Fetch was created after this code was originally written).

Reply:
Switched to `fetch()`